### PR TITLE
Ability entity, DAO interface and its implementation, AbilityType enum

### PR DESCRIPTION
**Pull request by AdamDobias**

### CODE REVIEW

File: src/main/java/fi/muni/pa165/hauntedhouses/entity/Ability.java [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+
+    public Set<Ghost> getGhosts() {
+        return Collections.unmodifiableSet(ghosts);
+    }
+
+    public void setGhosts(Set<Ghost> ghosts) {
+        this.ghosts = ghosts;
+    }
+
+
```
Comment (klarakufova) [inline src/main/java/fi/muni/pa165/hauntedhouses/entity/Ability.java]: Could you please add `equals` and `hashCode`?

File: src/main/java/fi/muni/pa165/hauntedhouses/enums/AbilityType.java [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+ * and open the template in the editor.
+ */
+package fi.muni.pa165.hauntedhouses.enums;
+
+/**
+ *
+ * @author Dubak
+ */
+public enum AbilityType {
+
```
Comment (klarakufova) [inline src/main/java/fi/muni/pa165/hauntedhouses/enums/AbilityType.java]: Could you please add some items? For example some of the ones that were in the original class diagram, such as _FIRE_, _DARKNESS_, _WATER_.

File: src/main/java/fi/muni/pa165/hauntedhouses/enums/AbilityType.java [hunk 3]
```diff
@@ -0,0 +1,14 @@
+/*
+ * To change this license header, choose License Headers in Project Properties.
+ * To change this template file, choose Tools | Templates
+ * and open the template in the editor.
+ */
+package fi.muni.pa165.hauntedhouses.enums;
+
+/**
+ *
+ * @author Dubak
```
Comment (klarakufova) [inline src/main/java/fi/muni/pa165/hauntedhouses/enums/AbilityType.java]: Probably not your name? 🤣

Comment (TomyAngelo) [reply src/main/java/fi/muni/pa165/hauntedhouses/enums/AbilityType.java]: Also please delete first rows with license craps in every class.

File: src/main/java/fi/muni/pa165/hauntedhouses/dao/AbilityDao.java [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+     * @return ability specified by (unique) name, null otherwise
+     */
+    public Ability findByName(String abilityName);
+    
+    /**
+     * finds all abilities currently stored in DB
+     * @return list of currently stored abilities
+     */
+    public List<Ability> findAll();
+
```
Comment (TomyAngelo) [inline src/main/java/fi/muni/pa165/hauntedhouses/dao/AbilityDao.java]: Delete unnecessary empty rows on the end of every class.

**Pull request merged**